### ISSUE TITLE
False positive no-unused-vars for type imports

### ISSUE THREAD

**Issue by Haaxor1689** [typescript-eslint/typescript-eslint#1044]
**What code were you trying to parse?**

```ts
// foo.ts
export default class Foo {
    static foo;
}

// goo.ts
import Foo from "./foo"; // Foo marked as unused

const goo: Partial<Foo> = null;
// Foo.foo; // Uncommenting this line fixes the unused warning
```

**What did you expect to happen?**

There should be no warning about `Foo` being unused in the second file.

**What actually happened?**

The above example shows `no-unused-vars` warning on Foo import if the only usage is in type declaration only. Warning only goes away if the type is used f.e. as in the example code after uncommenting last line.

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `2.3.2` |
| `TypeScript`                | `3.7.0-beta` |
| `ESLint`                    | `6.5.1` |
| `node`                      | `10.16.3` |
| `npm`                       | `6.9.0` |

**Comment by bradzacher** [comment]
duplicate of #363
